sdk/nodejs/cmd/pulumi-language-nodejs/main.go

```diff
@@ -563,115 +544,130 @@ func (host *nodeLanguageHost) Run(ctx context.Context, req *pulumirpc.RunRequest
 			"It looks like the Pulumi SDK has not been installed. Have you run npm install or yarn install?")
 	}
 
+	// Channel producing the final response we want to issue to our caller. Will get the result of
+	// the actual nodejs process we launch, or any results caused by errors in our server/pipes.
+	responseChannel := make(chan *pulumirpc.RunResponse)
+	defer close(responseChannel)
```

Comment:
This will still panic if execNodeJs finishes after pipesDone or handle.Done return an error because that'll exit this function, closing this channel, and then the goroutine will attempt to write to a closed channel, panic, and kill the process.

This can be fixed by using a channel with a buffer of one and dropping the `defer close(responseChannel)`. Let the channel be GCed if this function returns with an error. This function is short-lived, so that's not a big deal.

Another alternative is to use the channel just to signal that execNodeJs is done, but store the value in a variable in the scope:

```go
var response *pulumirpc.RunRespnse
haveResponse := make(chan struct{})
go func() {
  defer close(haveResponse)
  response = ...
}()
```

This way, the channel will always be closed when execNodeJs finishes.

Reply:
Good spot, it looks like we can just move the close inside the goroutine to fix this as well?